bin/crew

```diff
@@ -1035,30 +1038,44 @@ def shrink_dir(dir)
   unless CREW_SHRINK_ARCHIVE == '0'
     Dir.chdir dir do
       if File.exist?("#{CREW_PREFIX}/bin/rdfind")
-        puts "Using rdfind to find duplicate or hard linked files."
-        system "#{CREW_PREFIX}/bin/rdfind -removeidentinode true -makesymlinks true -makeresultsfile false ."
+        puts "Using rdfind to convert duplicate files to hard links."
+        system "#{CREW_PREFIX}/bin/rdfind -removeidentinode true -makesymlinks false -makehardlinks true -makeresultsfile false ."
```

Comment:
Shouldn't we be removing duplicates instead?  The idea is to reduce clutter and save space, correct?

Reply:
This does remove duplicates. What behavior specifically are you not seeing that you would like to see?

Reply:
Hard linking solves the space issue because hard linked files don't take up any more space.

Symlinking the duplicate files ends up wreaking havoc because creating proper relative symlinks isn't trivial, and absolute symlinks get broken during package and install...

Reply:
I guess I'm wondering why there are any duplicates in the first place?  Why is it necessary to have hard links?

Reply:
Packages like git will link a bunch of apps to one binary.

I think another package which does this is BusyBox?

When we use upx or another compression program on such a binary, it will replace the binary with a compressed one, but break the hard link.

E.g.

```
ln git git2
upx --lzma git
```
(Now you have two files — a compressed git binary and an uncompressed git2 binary.)

However
```
ln git git2
cp git git.tmp
upx --lzma git.tmp
cp git.tmp git
rm git.tmp
```
Now you have a compressed `git` and `git2`, and both are hard linked to each other.